moveit_ros/moveit_servo/include/moveit_servo/servo_calcs.h

```diff
@@ -104,6 +104,12 @@ class ServoCalcs
    */
   void changeRobotLinkCommandFrame(const std::string& new_command_frame);
 
+  /** \brief Retrieve the status code */
+  StatusCode getStatus() const
+  {
+    return status_;
+  }
+
```

Comment:
I think there is a data race on the status_ variable if you do it this way. The reason being that the variable is being modified in a thread that executes mainCalcLoop whereas the getter can be (and will be) called from another thread or various different threads. However, the calculateSingleIteration function already has a ROS publisher for the Servo status so I think it would work just fine if the PoseTracking interface had a subscriber for that topic and would update it's own copy of the status information using that subscriber.